### PR TITLE
federated authentication documentation

### PR DESCRIPTION
**Pull request by j0hnmwangi**
# Description

Changes made in this pull request:

- Separated client and server certificate creation options.
- Documented a terraform example that can be used for federated authentication.
- Added images for illustration purposes.
- fixed terraform apply update in place for minimum example affecting cloudwatch log group.

## Features/Fixes/Patches/Changes list:

Please check the [CHANGELOG.md](https://github.com/boldlink/terraform-aws-client-vpn/blob/fix/federated-authentication/CHANGELOG.md#101---2023-09-14)

### PR Owners Checklist:
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you checked there aren't open [Pull Requests](../../../pulls) of upstream or parallel code that will cause conflicts?
* [x] Have you updated the `CHANGELOG.md`?
* [x] Have you updated the `README.md`(s)?
* [x] OWNER: Does your submission pass?
    * [ ] Pre-commit (attach logs/print screen to this PR)
    * [ ] Checkov/terrascan (attach logs/print screen to this PR)
    * [ ] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### PR Reviewers Checklists?
* [ ] Are all your comments/changes resolved?
* [ ] Are you happy with the OWNER checklists and additional information provided?
* [ ] Does your review tests pass?
    * [ ] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?

### CODE REVIEW

File: media/saml.png
Comment (mariomerco) [inline media/saml.png:1]: Does the background need to be transparent?

File: media/client-vpn-certificate.png
Comment (mariomerco) [inline media/client-vpn-certificate.png:1]: Isn't the Client VPN endpoint inside the region? regardless is one of our resources or not?

**Review by mariomerco — COMMENTED** [review]
Added a couple of comments/questions in the diagrams, the rest looks good to me!

File: media/saml.png
Comment (j0hnmwangi) [reply media/saml.png:1]: I have updated the images used. Please have a look

File: media/client-vpn-certificate.png
Comment (j0hnmwangi) [reply media/client-vpn-certificate.png:1]: Yes. I have modified the image to have the endpoint inside the region.

File: media/saml.png
Comment (mariomerco) [reply media/saml.png:1]: Just a very minor one now hehehe Please, rename AWS SSO to **AWS Identity Center**. They changed that name a couple of months ago (and confused us lol).

**Review by mariomerco — DISMISSED** [review]
Super minor changed required (Rename "AWS SSO" with "AWS Identity Center"). Other than that, looks good to me! approved!

**Review by mariomerco — APPROVED** [review]
LGTM!

**Pull request merged**